src/lib/Embed.hs

```diff
@@ -715,6 +741,25 @@ indexToIntE ty idx = case ty of
     imul rSize lIdx >>= iadd rIdx
   TC (IntRange _ _)     -> indexAsInt idx
   TC (IndexRange _ _ _) -> indexAsInt idx
+  RecordTy types -> do
```

Comment:
BTW is there any reason to keep the record type as an n-ary constructor instead of a binary one, like we do for tuples?

Reply:
By binary constructor do you mean something like `RecordTy Label Type Type`, and expressing records as something like `RecordTy "foo" x (RecordTy "bar" y (Unit))`? The main reasons I went with this over that are:
- I intended the types `{a:Int & b:Real}` and `{b:Real & a:Int}` to be the same. If we used a binary constructor, we would have to worry about the labels not being in the right order, and possibly need to do list sorting to check that things match. Using a single n-ary constructor lets you easily enforce a single order internally while allowing arbitrary orders in the parsing and typechecking logic.
- The "rest" argument in a binary constructor needs to be another `RecordTy` instance, but I don't think the type system can support that. For polymorphic extensible records we probably need to introduce a new kind for a row of types, but that didn't seem worth the effort in the short term. (Tuples don't have this problem because `(a, b)` is a valid type no matter what `b` is, but `{foo:a, ...b}` only makes sense if `b` is a record type.)